influxdb_client/client/influxdb_client.py

```diff
@@ -216,13 +228,16 @@ def write_api(self, write_options=WriteOptions(), point_settings=PointSettings()
         """
         return WriteApi(influxdb_client=self, write_options=write_options, point_settings=point_settings)
 
-    def query_api(self) -> QueryApi:
+    def query_api(self, query_options: QueryOptions = QueryOptions()) -> QueryApi:
         """
         Create a Query API instance.
 
+        :param query_options: optional query api configuration
         :return: Query api instance
         """
-        return QueryApi(self)
+        if self.profilers is not None:
+            query_options.profilers = self.profilers
```

Comment:
This will override the setting from `query_options` parameter:

```python
client = InfluxDBClient(url=self.host, token=self.auth_token, debug=self.debug, org=self.org, profilers=["query"])

...

query_api = client.query_api(query_options=QueryOptions(profilers=["query", "operator"]))
```

The `query_api` will use: `profilers=["query"]`, but the user specified: `profilers=["query", "operator"]`.